howdoi/howdoi.py

```diff
@@ -621,7 +623,16 @@ def howdoi(raw_query):
         cache.set(cache_key, res)
     except (RequestsConnectionError, SSLError):
         res = {'error': f'Unable to reach {args["search_engine"]}. Do you need to use a proxy?\n'}
-
+    except BlockError:
+        BLOCKED_ENGINES.append(args['search_engine'])
+        if BLOCKED_ENGINES == SUPPORTED_SEARCH_ENGINES:
```

Comment:
I don't believe that this equality check will work between `BLOCKED_ENGINES`, a list and `SUPPORTED_SEARCH_ENGINES`, a tuple.

```
>>> type((1, 2, 3))
<class 'tuple'>
>>> type([1, 2, 3])
<class 'list'>
>>> (1, 2, 3) == [1, 2, 3]
False
```

However, if you made both of these objects a `Set` then the equality will work

```
>>> set((1, 2, 3)) == set([1, 2, 3])
True
```

You can probably change to

```
SUPPORTED_SEARCH_ENGINES = set(['google', 'bing', 'duckduckgo'])
BLOCKED_ENGINES = set()
```

rather than doing the conversion only when you do the comparison.

Reply:
Note - look at my comment after this - we might be able to remove the equality check altogether.

Reply:
You're right, it's not going to work with sets and lists. I've updated the method so that it doesn't use the if statement here, using the next() function instead as suggested below.